crafter-search-batch-indexer/src/main/java/org/craftercms/search/batch/impl/AbstractBinaryFileWithMetadataBatchIndexer.java

```diff
@@ -457,6 +466,13 @@ protected abstract void doUpdateContent(final String indexId, final String siteN
                                             final Resource toResource, final UpdateDetail updateDetail,
                                             final UpdateStatus updateStatus);
 
+    protected Map<String, Object> collectRemoteAssetMetadata(String binaryPath) {
+        Map<String, Object> metadata = new HashMap<>();
+        String internalName = FilenameUtils.getName(binaryPath);
+        metadata.put(internalNameFieldName, internalName);
```

Comment:
Why not put this as part of the `extractMetadata` method instead of a new method?

Reply:
Hi @avasquez614 What `extractMetadata` method are you mentioning for this?

* If the `extractMetadata` in the same class: https://github.com/craftercms/search/blob/develop/crafter-search-batch-indexer/src/main/java/org/craftercms/search/batch/impl/AbstractBinaryFileWithMetadataBatchIndexer.java#L460: I think this method has a different meaning to extract metadata for the jacket in `/site/documents`. And it is only used when we update a binary associated with the jacket. While updating any other binary, this method is not called at all.

* If the `extractMetadata` in Tika parser: https://github.com/craftercms/search/blob/develop/crafter-search-elasticsearch/src/main/java/org/craftercms/search/elasticsearch/impl/tika/TikaDocumentParser.java#L118: This would also affect other indexers such as preview. `internalName` seems only being used in the `authoring` indexer. However, if we want to add this to all the indexers, then this method could be a candidate. Or we can put it in `AbstractDocumentBuilder` as we have been doing for some fields such as `localId`, `rootId`: https://github.com/craftercms/search/blob/develop/crafter-search-commons/src/main/java/org/craftercms/search/commons/service/impl/AbstractDocumentBuilder.java#L127

In terms of the implementation, the `collectRemoteAssetMetadata` is aligned with what the method `collectMetadata` is doing for a local binary here: https://github.com/craftercms/search/blob/develop/crafter-search-batch-indexer/src/main/java/org/craftercms/search/batch/impl/AbstractBinaryFileWithMetadataBatchIndexer.java#L443.

Reply:
I understand the purpose of the method now, thanks